### PR TITLE
[dockers] Prevent apt-get from installing suggested and recommended packages by default

### PR DESCRIPTION
**Pull request by jleveque**
**- What I did**

On April 25, the Debian Jessie base image we download from the Docker repository was updated. This update caused a change in package dependencies, which caused a conflict in the DHCP relay docker. A fix was checked in via PR https://github.com/Azure/sonic-buildimage/pull/1659. However, this fix is more of a workaround, because previously, isc-dhcp-client was not installed in any Docker container, and now it is being installed in all Docker containers unnecessarily.

The root cause of this issue was a change in some "suggested" and "recommended" packages which caused the following chain reaction:

We install perl in docker-base. Apt-get now lists netbase as a "recommended" package of perl; netbase lists ifupdown as a "recommended" package; ifupdown lists isc-dhcp-client as a "recommended" package.

Thus, we were now installing netbase, ifupdown and isc-dhcp-client (and other unnecessary packages) in the docker-base image. Our apt-get configuration was installing recommended packages by default, which caused the installation of these unnecessary packages. This PR configures apt-get to *not* install "recommended" or "suggested" packages by default. All necessary packages should be installed explicitly. This PR not only fixes the isc-dhcp-client conflict, but should also help keep the image lean.

Once this PR is merged, we can revert PR https://github.com/Azure/sonic-buildimage/pull/1659.

After making this change, I also needed to adjust the installation of dependencies in docker-fpm-quagga, docker-snmp-sv2 and docker-sonic-vs. In doing so, I also aligned the respective Dockerfile.j2 templates with the cleaner format I have been using recently.

Resolves https://github.com/Azure/sonic-buildimage/issues/1669

### CODE REVIEW

**Comment by jleveque** [comment]
@zhenggen-xu: Please review and test this PR while reverting your previous PR (https://github.com/Azure/sonic-buildimage/pull/1659).

**Comment by lguohan** [comment]
virtual switch test failed.

File: dockers/docker-base/no_install_recommend_suggest [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Instruct apt-get to NOT install "recommended" or "suggested" packages by
+# default when installing a package.
+
+APT::Install-Recommends "false";
```
Comment (qiluo-msft) [inline dockers/docker-base/no_install_recommend_suggest]: Consider merge with files/apt/apt.conf.d/81norecommends, and share the same file in both host and docker images.

Comment (jleveque) [reply dockers/docker-base/no_install_recommend_suggest]: Good idea.

**Pull request merged**